Boa.Constrictor.RestSharp/Abilities/AbstractRestSharpAbility.cs

```diff
@@ -86,7 +83,15 @@ protected AbstractRestSharpAbility(RestClient client)
         /// Adds a cookie to the RestSharp client.
         /// </summary>
         /// <param name="cookie">The cookie to add to the RestSharp client.</param>
-        public void AddCookie(Cookie cookie) => Client.CookieContainer.Add(cookie);
+        public void AddCookie(Cookie cookie)
+        {
+            if (Client.Options.CookieContainer == null)
```

Comment:
I have a question about the new `CookieContainer`. How is it initialized? Is it an object that will automatically be created when the `RestClient` is created?

Reply:
Like most properties that were previously under `RestClient`, `CookieContainer` is now part of `RestClientOptions` and passed to the client on initialization.

For example


```
var cookies = new CookieContainer()
cookies.Add("MyCookie", "CookieValue");


var options = new RestClientOptions
{
  CookieContainer = cookies 
}


var client = new RestClient(options);

```


The problem is when accessing `client.Options` you only get access to `ReadOnlyRestClientOptions` so you can't initialize the `CookieContainer` at that point

Reply:
Got it! That makes sense. That is a better design.